lib/arelastic/builders/query_builder.rb

```diff
@@ -0,0 +1,48 @@
+module Arelastic
+  module Builders
+    module QueryBuilder
+      MACROS_TO_ARELASTIC = {
+        bool:         Arelastic::Queries::Bool,
+        ids:          Arelastic::Queries::Ids,
+        match_all:    Arelastic::Queries::MatchAll,
+        match_none:   Arelastic::Queries::MatchNone,
+        match_phrase: Arelastic::Queries::MatchPhrase,
+        prefix:       Arelastic::Queries::Prefix,
+        term:         Arelastic::Queries::Term,
+        terms:        Arelastic::Queries::Terms
```

Comment:
It might be worthwhile to add explicit tests for these similar to builder tests on  https://github.com/matthuhiggins/arelastic/blob/master/test/arelastic/builders/search_test.rb.

While these methods are defined declaratively. it would be pretty easy to unintentionally override them with the addition of new methods on `Arelastic::Builders::Search`.

Reply:
I am a little concerned about adding to a builder interface that can't support [all queries](https://github.com/matthuhiggins/arelastic/blob/fc896a0c7469c2f2a07a879915012c14d9e74186/lib/arelastic/queries.rb) including `query` and `filter`.

Reply:
You can't define `query` and `filter` on this class because those methods exist.

We could create `Model.arelastic.query_builder`? Then use it as `query_builder.filter(...)` etc.